fairlearn/adversarial/_preprocessor.py

```diff
@@ -151,7 +103,7 @@ def fit(self, X, y=None):
             # to the transformer.
             X_temp = self._check(X, init=True)
             self.n_features_in_ = X_temp.shape[0]
-            if _get_type(X_temp, self.dist_assumption) == "category":
+            if type_of_target(X_temp) in ["multiclass", "multilabel-indicator"]:
```

Comment:
@adrinjalali I am having doubts about this line here. Should it maybe be only `multiclass`? I am looking at the file again now for another test and there is the comment about `multilabel indicator` not needing to be encoded, line 98 above . I initially understood it as in not needing to be one hot encoded, but maybe it doesn't need to be handled at all here? I am not sure.

Reply:
The question is also that this seems to be not tested in the first place. Writing a test for both cases would make it clearer whether we need it or not I guess.

Reply:
I had to review all of it now for another test and I see a few testing gaps, especially since `get_type` is being removed. I might have to think a bit how to replace those tests with something less hardcoded type-wise, but still complete. I have another branch in progress on top of this one with further fixes, but I'll have that wait a bit, I'll focus on this first.